### PR TITLE
Read book publisher

### PR DESCRIPTION
**Pull request by piotr921**

### CODE REVIEW

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void success(Supplier<ResultModel> resultModelSupplier) {
                 addBooksToResultList(resultModelSupplier.get().getBookData());
                 firstPaginator = resultModelSupplier.get().getPaginator();
 
-                if (isFinished()){
+                if (isFinished()) {
```
Comment (mprtcz) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java:62]: :+1:

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java [hunk 2]
```diff
@@ -101,8 +101,8 @@ public void failed(String message) {
         }
     }
 
-    private void checkIfFinished(){
-        if (isFinished()){
+    private void checkIfFinished() {
+        if (isFinished()) {
```
Comment (mprtcz) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java:105]: :+1:

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/paginator/impl/DefaultPaginator.java [hunk 3]
```diff
@@ -7,8 +7,8 @@
  */
 public class DefaultPaginator implements Paginator{
 
-    private final int defaultNumberOfPages = 1;
-    private final int defaultItemsOnPage = 20;
+    private static final int defaultNumberOfPages = 1;
```
Comment (mprtcz) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/paginator/impl/DefaultPaginator.java:10]: Bad naming convention

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java [hunk 4]
```diff
@@ -68,7 +68,7 @@ public void success(Supplier<ResultModel> resultModelSupplier) {
 
             @Override
             public void failed(String message) {
-                addBooksToResultList(Collections.EMPTY_LIST);
+                addBooksToResultList(Collections.emptyList());
```
Comment (agawda) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/impl/GoogleCrawler.java:71]: 👍

File: googlecrawler/src/main/java/com/blackbook/googlecrawler/paginator/impl/DefaultPaginator.java (hunk 3 above)
Comment (agawda) [inline googlecrawler/src/main/java/com/blackbook/googlecrawler/paginator/impl/DefaultPaginator.java:10]: Capitalize.

File: taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReader.java [hunk 5]
```diff
@@ -14,33 +15,68 @@
...
+                        .authors(readAuthors(detailsPage))
+                        .bookPageUrl(detailsUrl)
+                        .coverUrl(readCoverUrl(detailsPage))
                         .build())
                 .build();
     }
 
-    private String readTitle(Element book) {
-        return book
+    private String readCoverUrl(Document detailsPage) {
```
Comment (mprtcz) [inline taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/PromoDetailsReader.java:38]: :+1:  for creating small separate methods

File: taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/Scraper.java [hunk 6]
```diff
@@ -39,7 +39,7 @@ public void start(CrawlerActionListener actionListener, ExecutorService executor
             String promotionPageUrl = String.format(PROMOTION_PAGE_URL, pageId);
             Document document = connector.getDocumentFromWebPage(promotionPageUrl);
             Elements books = document.select(".product-container");
-            books.forEach(book -> discountData.add(detailsReader.readDiscountDataProperties(book)));
+            books.forEach(book -> discountData.add(detailsReader.readDiscountDataProperties(connector, book)));
```
Comment (mprtcz) [inline taniaksiazkascraper/src/main/java/com/blackbook/taniaksiazkascraper/scraper/Scraper.java:42]: :+1: for inversion

**Review by mprtcz — APPROVED** [review]
Take a look at my suggestions

**Review by agawda — APPROVED** [review]

**Pull request merged**